packages/flutter/lib/src/widgets/view.dart

```diff
@@ -106,13 +145,656 @@ class View extends StatelessWidget {
+        'pipelineOwner: $pipelineOwner, '
+        'renderViewRepository: $renderViewRepository'
+      ')';
+  }
+}
+
+class _MultiChildComponentWidget extends Widget {
+  const _MultiChildComponentWidget({
+    super.key,
+    List<Widget> views = const <Widget>[],
+    Widget? child,
+  }) : _views = views, _child = child;
+
+  // It is up to the subclasses to make the relevant properties public.
+  final List<Widget> _views;
+  final Widget? _child;
+
+  @override
+  Element createElement() => _MultiChildComponentElement(this);
+}
+
+/// A collection of sibling [View]s.
+///
+/// The widget can only be used in places were a [View] widget is legal. In
+/// practical terms, it can be used at the root of the widget tree outside of
+/// any [View] widget, as a child to a another [ViewCollection], or in the
+/// [ViewAnchor.view] slot of a [ViewAnchor] widget. It is not required to be a
+/// direct child of those widgets; other non-[RenderObjectWidget]s may appear
+/// in between the two.
+///
+/// Similarly, the [views] children of this widget must be [View]s, but they
+/// may be wrapped in additional non-[RenderObjectWidget]s (e.g.
+/// [InheritedWidget]s).
+class ViewCollection extends _MultiChildComponentWidget {
+  /// Creates a [ViewCollection] widget.
+  ///
+  /// The provided list of [views] must contain at least one widget.
+  const ViewCollection({super.key, required super.views}) : assert(views.length > 0);
+
+  /// The [View] descendants of this widget.
+  ///
+  /// The [View]s may be wrapped in other non-[RenderObjectWidget]s (e.g.
+  /// [InheritedWidget]s). However, no [RenderObjectWidget] is allowed to appear
+  /// between the [ViewCollection] and the next [View] widget.
+  List<Widget> get views => _views;
+}
+
+/// Decorates a [child] widget in a surrounding [View] with a side-[View].
+///
+/// This widget must have a [View] ancestor, into which the [child] widget
+/// is rendered.
+///
+/// Typically, a [View] or [ViewCollection] widget is used in the [view] slot to
+/// define the content of the side view(s). Those widgets may be wrapped in
+/// other non-[RenderObjectWidget]s (e.g. [InheritedWidget]s). However, no
+/// [RenderObjectWidget] is allowed to appear between the [ViewAnchor] and the
+/// next [View] widget in the [view] slot. The widgets in the [view] slot have
+/// access to all [InheritedWidget]s above the [ViewAnchor] in the tree.
+class ViewAnchor extends StatelessWidget {
+  /// Creates a [ViewAnchor] widget.
+  const ViewAnchor({
+    super.key,
+    this.view,
+    required this.child,
+  });
+
+  /// The widget that defines the view anchored to this widget.
+  ///
+  /// Typically, a [View] or [ViewCollection] widget is used, which may be wrapped
+  /// in other non-[RenderObjectWidget]s (e.g. [InheritedWidget]s).
+  final Widget? view;
+
+  /// The widget below this widget in the tree.
+  ///
+  /// It is rendered into the surrounding view, not in the view defined by
+  /// [view].
```

Comment:
This feels a bit abstract. Would it make sense to mention an example use case, like a button (child) with a tooltip (view)? Or perhaps this should be on the `ViewAnchor`'s comment above?

Reply:
Good idea, added the the tooltip example to make this a little more concrete.